src/service.js

```diff
@@ -0,0 +1,316 @@
+if(!global._babelPolyfill) { require('babel-polyfill'); }
+
+import Proto from 'uberproto';
+import mongoose from 'mongoose';
+import filter from 'feathers-query-filters';
+import errors from 'feathers-errors';
+import errorHandler from './error-handler';
+
+// Use native promises
+mongoose.Promise = global.Promise;
```

Comment:
Hmm. I agree, however if we are to enforce native promises and we want to keep the same API everywhere then we either need to document that the user should do this or get the user to pass in the mongoose object. Which one do you prefer?

Personally I think I like the first option better.